services/web-server/test/graphql/tasks_graphql_test.js

```diff
@@ -64,4 +68,81 @@ helper.secrets.mockSuite(testing.suiteName(), [], function(mock, skipping) {
       assert.equal(response.data.createTask.taskId, taskId);
     });
   });
+
+  suite('Task Subscriptions', function() {
+    helper.withMockedEventIterator();
+
+    const createSubscriptionClient = async () => {
+      return new Promise(function(resolve, reject) {
+        const subscriptionClient = new SubscriptionClient(
+          `ws://localhost:${helper.serverPort}/subscription`,
+          {
+            reconnect: true,
+          },
+          WebSocket,
+        );
+        subscriptionClient.onConnected(function() {
+          resolve(subscriptionClient);
+        });
+        subscriptionClient.onError(function(err) {
+          reject(err);
+        });
+      });
+    };
+
+    const getClient = (subscriptionClient) => {
+      const cache = new InMemoryCache();
+      const link = new WebSocketLink(subscriptionClient);
+
+      return new ApolloClient({ cache, link });
+    };
+
+    test('subscribe works', async function(){
+      // We need to create this subscription client separately so we can close it after our test
+      // Otherwise, our tests will just hang and timeout
+      let subscriptionClient = await createSubscriptionClient();
+      const client = getClient(subscriptionClient);
+      const task = helper.makeTaskDefinition();
+      const taskGroupId = task.taskGroupId;
+      const subscriptions = [
+        'tasksDefined',
+      ];
+
+      const payload = {
+        tasksSubscriptions: {
+          status: {
+            taskId: "subscribe-task-id",
+            taskGroupId: "subscribe-task-group-id",
+          },
+        },
+      };
+
+      const asyncIterator = new Object();
+      asyncIterator[Symbol.asyncIterator] = async function*() {
+        yield payload;
+      };
+
+      helper.SetNextAsyncIterator(asyncIterator);
+
+      let tasksSubscriptionsResult;
+      let taskSubscription = client.subscribe({
+        query: gql`${subscribeTasks}`,
+        variables: {
+          taskGroupId,
+          subscriptions,
+        },
+      }).subscribe(
+        (value) => tasksSubscriptionsResult = value,
+      );
+
+      //wait for event to be processed
+      await new Promise(resolve => setTimeout(resolve, 10));
```

Comment:
Timeouts are really dangerous in tests -- on a busy system, this might end up taking 11ms, in which case the test will fail.  But if you set a longer timeout, then the whole test suite starts to take forever.  A better approach here is to poll, waiting for the result.  We even have [a utility for that](https://github.com/taskcluster/taskcluster/tree/master/libraries/testing#poll)!
```suggestion
      await testing.poll(
        () => assert(tasksSubscriptionResult),
        100, 10);
```
this will try every 10ms for up to a second, then give up.